src/test/java/io/stargate/sgv2/jsonapi/service/processor/MeteredCommandProcessorTest.java

```diff
@@ -0,0 +1,131 @@
+
+            }
+          }
+          """;
+
+      CountDocumentsCommands countCommand =
+          objectMapper.readValue(json, CountDocumentsCommands.class);
+      CommandContext commandContext = new CommandContext("namespace", "collection");
+      CommandResult commandResult = new CommandResult(Collections.emptyList());
+      Mockito.when(commandProcessor.processCommand(commandContext, countCommand))
+          .thenReturn(Uni.createFrom().item(commandResult));
+      Mockito.when(stargateRequestInfo.getTenantId()).thenReturn(Optional.of("test-tenant"));
+      meteredCommandProcessor.processCommand(commandContext, countCommand).await().indefinitely();
+      String metrics = given().when().get("/metrics").then().statusCode(200).extract().asString();
+      List<String> httpMetrics =
+          metrics
+              .lines()
+              .filter(
+                  line ->
+                      line.startsWith("command_processor_process")
+                          && line.contains("error=\"false\""))
+              .toList();
+
+      assertThat(httpMetrics)
+          .satisfies(
+              lines -> {
+                assertThat(lines.size()).isEqualTo(3);
+                lines.forEach(
+                    line -> {
+                      assertThat(line).contains("command=\"CountDocumentsCommands\"");
+                      assertThat(line).contains("tenant=\"test-tenant\"");
+                      assertThat(line).contains("error=\"false\"");
+                      assertThat(line).contains("error_code=\"NA\"");
+                      assertThat(line).contains("error_class=\"NA\"");
+                      assertThat(line).contains("module=\"sgv2-jsonapi\"");
+                    });
+              });
+    }
+
+    @Test
+    public void errorMetrics() throws Exception {
+      String json =
+          """
+                                    {
+                                      "countDocuments": {
+
+                                      }
+                                    }
+                                    """;
+
+      CountDocumentsCommands countCommand =
+          objectMapper.readValue(json, CountDocumentsCommands.class);
+      CommandContext commandContext = new CommandContext("namespace", "collection");
+      Map<String, Object> fields = new HashMap<>();
+      fields.put("exceptionClass", "TestExceptionClass");
+      fields.put("errorCode", "TestErrorCode");
+      CommandResult.Error error = new CommandResult.Error("message", fields, Response.Status.OK);
+      CommandResult commandResult = new CommandResult(Collections.singletonList(error));
+      Mockito.when(commandProcessor.processCommand(commandContext, countCommand))
+          .thenReturn(Uni.createFrom().item(commandResult));
+      Mockito.when(stargateRequestInfo.getTenantId()).thenReturn(Optional.of("test-tenant"));
+      meteredCommandProcessor.processCommand(commandContext, countCommand).await().indefinitely();
```

Comment:
nit: formatting json, clear separation between test setup, execution and verification/assertion would improve readability here..